### PR TITLE
Bug 1224629 - Create an end-to-end test that verifies you can save a note in the Notes app

### PR DESCRIPTION
**Pull request by ubermarklar**
Round 2, because I'm horrible at rebasing code.
Created a test for writing and saving a note. This time with changes suggested by Johan and Martijn

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4c3002edc962c53f4daca9ba77289427ba93015

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1fb06d335600c2afe566815cc4b95d40e5332dcf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=15306b8bd8a2a374531ad4d024a0c6fe63412df3

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/notes/test_notes_new_note.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+        # This test creates a note and verifies that by checking the text
+
+        note_text = 'I am a note!'
+
+        self.notes = Notes(self.marionette)
+        self.notes.launch()
+
+        main_menu = self.notes.write_and_save_note(note_text)
+        self.assertEquals(main_menu.first_note_title, note_text)
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/notes/test_notes_new_note.py]: Nit: `assertEquals` with an `s` [is deprecated](https://docs.python.org/2/library/unittest.html#deprecated-aliases). Use assertEqual instead.

File: tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py]: Nit: `__init__.py` is missing in `tests/python/gaia-ui-tests/gaiatest/apps/notes/`. Then, the test will fail with this error:
`ImportError: No module named notes.app`

File: tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import expected, By, Wait
+from gaiatest.apps.base import Base, PageRegion
+
+
+class Notes(Base):
+
+    name = 'Notes'
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py]: I don't think the `name` is necessary.

File: tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    def launch(self):
+        Base.launch(self)
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(expected.element_present(
+                *self._app_ready_locator))))
+
+    def write_and_save_note(self, text):
+        Wait(self.marionette).until(expected.element_displayed(
+           Wait(self.marionette).until(expected.element_present(*self._note_content_locator))))
+        note_content = self.marionette.find_element(*self._note_content_locator)
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py]: You could do here:
`
note_content = Wait(self.marionette).until(expected.element_present(*self._note_content_locator))
Wait(self.marionette).until(expected.element_displayed(note_content))
`

File: tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        self.marionette.find_element(*self._save_note).tap()
+        return NotesMainMenu(self.marionette)
+
+    @property
+    def first_note_title(self):
+        return self.marionette.find_element(*self._note_content_locator).text
+
+class NotesMainMenu(PageRegion):
+
+    name = 'NotesMainMenu'
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py]: This is not needed at all.

File: tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    _main_locator = (By.ID, 'main')
+    _note_content_locator = (By.ID, 'note-content')
+
+    def __init__(self, marionette):
+        PageRegion.__init__(self, marionette, marionette.find_element(*self._main_locator))
+        Wait(self.marionette).until(lambda m: self.root_element.rect['x'] == 0 and self.root_element.is_displayed())
+
+    @property
+    def first_note_title(self):
+        return self.marionette.find_element(*self._note_content_locator).text
```
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py]: Missing end line?

File: tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py (hunk 3 above)
Comment (ubermarklar) [inline tests/python/gaia-ui-tests/gaiatest/apps/notes/app.py]: When I removed `name = 'Notes'` the test failed to launch the notes application. When left in the test functioned properly and passed.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9b892c11926e1d6d14cdc8df8535f07b09363c8d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bf6d56641627436ffe162f592763a7faee725f50

**Pull request closed without merging**